operator/src/test/java/org/keycloak/operator/testsuite/integration/ServiceMonitorTest.java

```diff
@@ -61,14 +69,17 @@ public void testServiceMonitorDisabledExplicitly() {
     }
 
     @Test
-    public void testServiceMonitorDisabledLegacyManagement() {
+    public void testServiceMonitorLegacyManagement() {
```

Comment:
The original implementation intentionally didn't create a ServiceMonitor for `legacy-observability-interface` as it is deprecated. If we're going to support this, should we create a separate issue (happy for it to be part of this PR still) so that the change gets some exposure in the release notes?